src/components/smart_objects/src/object_schema_item.cc

```diff
@@ -251,45 +252,45 @@ CObjectSchemaItem::CObjectSchemaItem(const Members& members)
 
 void CObjectSchemaItem::RemoveFakeParams(
     SmartObject& Object, const utils::SemanticVersion& MessageVersion) {
-  for (SmartMap::const_iterator it = Object.map_begin();
-       it != Object.map_end();) {
-    const std::string& key = it->first;
+  for (const auto& key : Object.enumerate()) {
     std::map<std::string, SMember>::const_iterator members_it =
         mMembers.find(key);
-    if (mMembers.end() == members_it
-        // FIXME(EZamakhov): Remove illegal usage of filed in AM
-        && key.compare(connection_key) != 0 && key.compare(binary_data) != 0 &&
-        key.compare(app_id) != 0) {
-      ++it;
-      Object.erase(key);
 
-    } else if (mMembers.end() != members_it &&
-               GetCorrectMember(members_it->second, MessageVersion)
-                   .mIsRemoved) {
-      ++it;
+    if (mMembers.end() != members_it) {
+      const SMember* member =
+          GetCorrectMember(members_it->second, MessageVersion);
+      if (!member || member->mIsRemoved) {
+        Object.erase(key);
```

Comment:
These changes cause invalid iterators. There's a reason that this used a normal for loop rather than a range-based one. See https://en.cppreference.com/w/cpp/container/map/erase

> References and iterators to the erased elements are invalidated.

Reply:
Ah, I see that this might not be the case, since `enumerate` seems to make a copy of the key list.